src/views/clusteroverview/overview/components/details-card/utils/types.ts

```diff
@@ -0,0 +1,206 @@
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+
+export type Descriptor<T = any> = {
+  path: string;
+  displayName: string;
+  description: string;
+  'x-descriptors'?: T[];
+  value?: any;
+};
+
+export type CRDDescription = {
+  name: string;
+  version: string;
+  kind: string;
+  displayName: string;
+  description?: string;
+  specDescriptors?: Descriptor[];
+  statusDescriptors?: Descriptor[];
+  resources?: {
+    name?: string;
+    version: string;
+    kind: string;
+  }[];
+};
+
+export type APIServiceDefinition = {
+  name: string;
+  group: string;
+  version: string;
+  kind: string;
+  deploymentName: string;
+  containerPort: number;
+  displayName: string;
+  description?: string;
+  specDescriptors?: Descriptor[];
+  statusDescriptors?: Descriptor[];
+  resources?: {
+    name?: string;
+    version: string;
+    kind: string;
+  }[];
+};
+
+export enum InstallModeType {
+  InstallModeTypeOwnNamespace = 'OwnNamespace',
+  InstallModeTypeSingleNamespace = 'SingleNamespace',
+  InstallModeTypeMultiNamespace = 'MultiNamespace',
+  InstallModeTypeAllNamespaces = 'AllNamespaces',
+}
+
+export type ClusterServiceVersionIcon = { base64data: string; mediatype: string };
+
+export enum ClusterServiceVersionPhase {
+  CSVPhaseNone = '',
+  CSVPhasePending = 'Pending',
+  CSVPhaseInstallReady = 'InstallReady',
+  CSVPhaseInstalling = 'Installing',
+  CSVPhaseSucceeded = 'Succeeded',
+  CSVPhaseFailed = 'Failed',
+  CSVPhaseUnknown = 'Unknown',
+  CSVPhaseReplacing = 'Replacing',
+  CSVPhaseDeleting = 'Deleting',
+}
+
+export enum CSVConditionReason {
+  CSVReasonRequirementsUnknown = 'RequirementsUnknown',
+  CSVReasonRequirementsNotMet = 'RequirementsNotMet',
+  CSVReasonRequirementsMet = 'AllRequirementsMet',
+  CSVReasonOwnerConflict = 'OwnerConflict',
+  CSVReasonComponentFailed = 'InstallComponentFailed',
+  CSVReasonInvalidStrategy = 'InvalidInstallStrategy',
+  CSVReasonWaiting = 'InstallWaiting',
+  CSVReasonInstallSuccessful = 'InstallSucceeded',
+  CSVReasonInstallCheckFailed = 'InstallCheckFailed',
+  CSVReasonComponentUnhealthy = 'ComponentUnhealthy',
+  CSVReasonBeingReplaced = 'BeingReplaced',
+  CSVReasonReplaced = 'Replaced',
+  CSVReasonCopied = 'Copied',
+}
+
+export type RequirementStatus = {
+  group: string;
+  version: string;
+  kind: string;
+  name: string;
+  status: string;
+  uuid?: string;
+};
+
+export type K8sResourceKind = K8sResourceCommon & {
```

Comment:
this can be useful to everyone maybe we should consider taking it to src/utils at some point :)

Reply:
I think that we should move it to a shared location as soon as it's needed in at least one other place.